### PR TITLE
Sentry Improvements

### PR DESCRIPTION
**Pull request by rh-glob**
- Add `X-Request-User` as sentry user.id and make configurable in settings
- Add more tags to be pulled from opaque and make configurable in settings
- Add ability to override the `before_send` function in the app using the library
- Add docs
- un-redact some safe cases in default `before_send`

### CODE REVIEW

File: microcosm_pubsub/sentry.py [hunk 1]
```diff
@@ -12,19 +18,33 @@
...
+
+DEFAULT_TAG_MAPPING = {
+    "X-Request-Id": "x-request-id",
+    "message_id": "message-id",
+    "media_type": "media-type",
+    "span_name": "span",
+    "uri": "uri",
+    "handler": "handler",
+}
+DEFAULT_USER_ID_KEY = "X-Request-Id"
```
Comment (tonybajan-globality) [inline microcosm_pubsub/sentry.py:32]: X-Request-User?

File: microcosm_pubsub/sentry.py [hunk 2]
```diff
@@ -33,41 +53,80 @@ def before_send(event, hint):
...
     dsn=None,
+    custom_tags_mapping=None,
+    custom_before_send_func=None,
+    custom_user_id=None,
 )
 @logger
-def configure_sentry(graph: ObjectGraph):
-    enabled, dsn, sentry = False, None, None
+def configure_sentry(graph: ObjectGraph) -> SentryConfig:
+    enabled, dsn, sentry,  = False, None, None
```
Comment (simpleigh-globality) [inline microcosm_pubsub/sentry.py:81]: Nit: inconsistent use of trailing commas (and a bonus space).

Comment (rh-glob) [reply microcosm_pubsub/sentry.py:81]: 🦅 👁️ @simpleigh-globality 👍

File: microcosm_pubsub/sentry.py [hunk 3]
```diff
@@ -33,41 +53,80 @@ def before_send(event, hint):
...
                     elif var_name.endswith("_id"):
                         continue
+                    elif value.startswith("http"):
+                        continue
+                elif isinstance(value, dict):
+                    for key in value.keys():
+                        if key.endswith("_id"):
+                            continue
+                        value[key] = "<redacted>"
+                    continue
```
Comment (tonybajan-globality) [inline microcosm_pubsub/sentry.py:67]: can we also let though anything in `{None, [], {}, set(), ""}` - definitely not secret, really helpful for debugging to know that something was unexpectedly empty.

File: microcosm_pubsub/sentry.py [hunk 4]
```diff
@@ -33,41 +53,80 @@ def before_send(event, hint):
...
         dsn=dsn,
         enabled=enabled,
-        client=sentry
+        client=sentry,
+        tag_mapping=tag_mapping,
+        user_id_key=custom_user_id_key
     )
+
+
+def import_custom_before_send(graph: ObjectGraph) -> Optional[Callable[[Any, Any], Any]]:
```
Comment (simpleigh-globality) [inline microcosm_pubsub/sentry.py:119]: Is there a test for this?

File: microcosm_pubsub/sentry.py [hunk 5]
```diff
@@ -33,41 +53,80 @@ def before_send(event, hint):
...
+    func_path = graph.config.sentry_logging.custom_before_send_func
```
Comment (tonybajan-globality) [inline microcosm_pubsub/sentry.py:120]: can the before send func be a binding instead? this seems like it's a good usecase for our DI framework

Comment (rh-glob) [reply microcosm_pubsub/sentry.py:120]: Ah good point! Will investigate

File: README.md [hunk 6]
```diff
@@ -129,3 +129,35 @@ the usual boilerplate:
...
+    },
+    "custom_user_id": "my-opaque-user-id"
+  }
+}
+```
+
+- `sentry_logging.enabled` - boolean
+- `sentry_logging.dsn` - url - the url of the project that errors will be
+ sent to
+- `sentry_logging.custom_before_send_func`  - python module path - allows
```
[the remark this answers is not in the document]
Comment (rh-glob) [reply README.md]: Updated to allow overriding in a more microcosm friendly way

**Pull request merged**